envoy/http/filter.h

```diff
@@ -174,11 +174,20 @@ enum class FilterTrailersStatus {
 
 /**
  * Return codes for encode metadata filter invocations. Metadata currently can not stop filter
- * iteration.
+ * iteration except in the case of local replies.
  */
 enum class FilterMetadataStatus {
-  // Continue filter chain iteration.
+  // Continue filter chain iteration for metadata only. Does not unblock returns of StopIteration*
+  // from (decode|encode)(Headers|Data).
   Continue,
+
+  // Continue filter chain iteration. If headers have not yet been sent to the next filter, they
+  // will be sent first via (decode|encode)Headers().
+  ContinueAll,
+
+  // Stops iteration of the entire filter chain. Only to be used in the case of sending a local
+  // reply from (decode|encode)Metadata.
+  StopIterationForLocalReply,
```

Comment:
Why is it this specific vs. just StopIteration? Couldn't theoretically someone right a filter in which they stop iteration and then release the metadata (or maybe in the future change it) when they get headers/data?

Reply:
It's this specific because I didn't want to implement FilterMetadataStatus::StopIteration in its full generality yet. If you think it's necessary at this time, I can add it.

Reply:
There's ASSERTs in the code to enforce that this is only returned in cases of local reply...